src/PerformanceSensitiveAnalyzers/UnitTests/CallSiteImplicitAllocationAnalyzerTests.cs

```diff
@@ -149,5 +149,97 @@ public void Foo()
                 // Test0.cs(12,9): warning HAA0102: Non-overridden virtual method call on a value type adds a boxing or constrained instruction
                 VerifyCS.Diagnostic(CallSiteImplicitAllocationAnalyzer.ValueTypeNonOverridenCallRule).WithLocation(12, 9));
         }
+
+        [Fact]
+        public async Task CallSiteImplicitAllocation_ObjectParamsCalledWithStringArray_NoWarning()
+        {
+            var sampleProgram = @"
+using System.IO;
+using Roslyn.Utilities;
+
+public class MyClass
+{
+    [PerformanceSensitive(""uri"")]
+    public void Testing()
+    {
+        string[] args = new [] { ""foo"", ""bar""};
+        string formatted = string.Format(""{0} {1}"", args);
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(sampleProgram);
+        }
+
+        [Fact]
+        public async Task CallSiteImplicitAllocation_CallingConstructorWithParams_Warning()
+        {
+            var sampleProgram = @"
+using System.IO;
+using Roslyn.Utilities;
+
+public class MyClass
+{
+    [PerformanceSensitive(""uri"")]
+    public MyClass()
+        : this(""foo"", ""bar"")
+    {
+    }
+
+    public MyClass(params object[] args)
+    {
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(sampleProgram,
+                VerifyCS.Diagnostic(CallSiteImplicitAllocationAnalyzer.ParamsParameterRule).WithLocation(9, 9));
```

Comment:
Please do not use this `WithLocation` overload instead use one of the following options:
- (preferred option when no arguments) the markup syntax (e.g. `[|MyClass|]` and no need for diagnostic)
- (preferred option when arguments) the markup syntax (e.g. `{|#0:MyClass|}` and `WithLocation(0)`)
- `WithSpan`